### PR TITLE
Enable parallelism in cudf & dask-cudf pytests

### PR DESCRIPTION
**Pull request by galipremsagar**
This PR enables pytest-xdist in cudf and dask-cudf pytests.

### CODE REVIEW

File: ci/test/cudf.sh
Comment (galipremsagar) [inline ci/test/cudf.sh:37]: Since the CI jobs have 8vCPUs, I've left out 2 cores for pytest to not choke.

Comment (galipremsagar) [reply ci/test/cudf.sh:37]: cc: @raydouglass @mike-wendt

**Comment by galipremsagar** [comment]
I've locally tested pytests with enabling cuda MPS, there isn't any noticeable speeds just by using MPS but however it would help us with CUDA context switching when there are multiple processes. 

@rapidsai/ops-codeowners Could you point where the following set of commands need to be added in this repo? These set of commands except running pytests need to be executed as a root user.

# Execute as root

export CUDA_VISIBLE_DEVICES="0"
nvidia-smi -i 0 -c EXCLUSIVE_PROCESS
nvidia-cuda-mps-control -d


# Run Pytests
...
...

# MPS Cleanup

echo quit | nvidia-cuda-mps-control
nvidia-smi -i 0 -c DEFAULT

**Comment by kkraus14** [comment]
Can we handle MPS in a follow up? Is there a net benefit without MPS?

**Comment by galipremsagar** [comment]
> Can we handle MPS in a follow up? Is there a net benefit without MPS?

Sure this can be done as a follow-up and maybe we can add MPS if we see context issues but currently I did not notice any difference in execution times. I've been using parallel pytests without enabling MPS for quite a very long time. So parallel pytests should run properly without enabling MPS too.

**Comment by raydouglass** [comment]
I don't think we can enable MPS as running commands as root is not possible inside of the gpuCI docker containers.

**Comment by galipremsagar** [comment]
> running commands as root is not possible inside of the gpuCI docker containers.

Didn't know that. In that case, we probably go ahead without MPS too.

**Comment by mike-wendt** [comment]
> I don't think we can enable MPS as running commands as root is not possible inside of the gpuCI docker containers.

Due to the security concerns this is not an option in gpuCI. If there is a way to do this without `root` access we can make it work; otherwise, we can only run things are a privileged user to limit system access as these run on a shared environment

**Comment by kkraus14** [comment]
IIRC this is a driver level setting that interacts with the kernel so you can't really run it within a container even with root unless you use the `--privileged` flag which is an obvious no-no.

Lets move forward without MPS for now and can explore it in the future.

**Comment by mike-wendt** [comment]
rerun tests

**Comment by mike-wendt** [comment]
Blocked until `runtime` images are fixed

**Comment by mike-wendt** [comment]
rerun tests

**Comment by mike-wendt** [comment]
rerun tests

**Review by mike-wendt — APPROVED** [review]

**Pull request merged**
